app/validation/validator.py

```diff
@@ -168,6 +178,35 @@ def validate_routing_on_answer_options(self, block, answer):
                 answer_errors.append(self._error_message(unrouted_error))
         return answer_errors
 
+    @staticmethod
+    def validate_when_rule(when_clause, existing_ids):
```

Comment:
`validate_when_rule` implies that it'll evaluate any `when` rule. But actually this method will error if you pass any `when` rule it at that doesn't contain an `id` attribute.

I'd suggest either checking that `id` is in each rule as part of this method or calling this method something like `validate_answer_id_when_rule`. I prefer the former.

Reply:
Good call on adding this check though 👍